include/swift/SIL/SILConstants.h

```diff
@@ -146,14 +148,20 @@ class SymbolicValue {
     AggregateSymbolicValue *aggregate;
   } value;
 
-  union {
-    UnknownReason unknown_reason;
+public:
+  /// When the value is Unknown, this contains information about the unfoldable
+  /// part of the computation.
+  struct UnknownInfo {
+    /// The value that was unfoldable.
+    SILNode *node;
 
-    // unsigned integer_bitwidth;
-    // ...
-  } aux;
+    /// A more explanatory reason for the value being unknown.
+    UnknownReason reason;
+
+    /// The call stack while evaluating the unknown value.
+    llvm::SmallVector<SourceLoc, 4> call_stack;
```

Comment:
The objects associated with SymbolicValue are allocated in the bump pointer allocator ... so their destructors are never called.  This means they cannot contain non-pod values like smallvector (it would leak memory when it contained values larger than its "small" size of 4).  Please switch this to use the same pattern as the AggregateValue.  You should use a trailing object allocation of the SourceLoc objects.  This will make the whole thing POD and in a continuous allocation.

Reply:
Also, please declare that class in the .cpp file, so the public interface of SymbolicValue is kept small.